internal/containerupdate/exec_updater_test.go

```diff
@@ -0,0 +1,124 @@
+package containerupdate
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/windmilleng/tilt/internal/k8s"
+	"github.com/windmilleng/tilt/internal/model"
+	"github.com/windmilleng/tilt/internal/testutils"
+)
+
+var toDelete = []string{"/foo/delete_me", "/bar/me_too"}
+var (
+	cmdA = model.Cmd{Argv: []string{"a"}}
+	cmdB = model.Cmd{Argv: []string{"b", "bar", "baz"}}
+)
+var cmds = []model.Cmd{cmdA, cmdB}
+
+func TestUpdateContainerDoesntSupportRestart(t *testing.T) {
+	f := newExecFixture(t)
+
+	err := f.ecu.UpdateContainer(f.ctx, TestDeployInfo, newReader("boop"), toDelete, cmds, false)
+	if assert.NotNil(t, err, "expect Exec UpdateContainer to fail if !hotReload") {
+		assert.Contains(t, err.Error(), "ExecUpdater does not support `restart_container()` step")
+	}
+}
+
+func TestUpdateContainerDeletesFiles(t *testing.T) {
+	f := newExecFixture(t)
+
+	// No files to delete
+	err := f.ecu.UpdateContainer(f.ctx, TestDeployInfo, newReader("boop"), nil, cmds, true)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	for _, call := range f.kCli.ExecCalls {
+		if len(call.Cmd) >= 1 && call.Cmd[0] == "rm" {
```

Comment:
this seems like a good use for that sliceutils.StringSliceStartsWith function that someone just added!